go/vt/discovery/healthcheck.go

```diff
@@ -463,7 +463,7 @@ func (hcc *healthCheckConn) processResponse(hc *HealthCheckImpl, shr *querypb.St
 		serving = false
 	}
 
-	if shr.TabletAlias != nil && *shr.TabletAlias != *oldTs.Tablet.Alias {
+	if shr.TabletAlias != nil && oldTs.Tablet.Alias.Uid != 0 && !proto.Equal(shr.TabletAlias, oldTs.Tablet.Alias) {
```

Comment:
Please add a comment here.

Checking for Uid != 0 is non-trivial code. Any reader of the code will not understand why you make an exception here.